builder/builder-next/controller.go

```diff
@@ -164,3 +174,44 @@ func newController(rt http.RoundTripper, opt Opt) (*control.Controller, error) {
 		// TODO: set ResolveCacheExporterFunc for exporting cache
 	})
 }
+
+func getGCPolicy(conf config.BuilderConfig, root string) ([]client.PruneInfo, error) {
+	var (
+		defaultKeepStorage int64
+		err                error
+	)
+
+	if conf.GC.DefaultKeepStorage != "" {
+		defaultKeepStorage, err = units.RAMInBytes(conf.GC.DefaultKeepStorage)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not parse '%s' as Builder.GC.DefaultKeepStorage config", conf.GC.DefaultKeepStorage)
+		}
+	}
+
+	var gcPolicy []client.PruneInfo
+	if conf.GC.Enabled {
+		if conf.GC.Policy != nil {
+			gcPolicy = make([]client.PruneInfo, len(conf.GC.Policy))
+			for i, p := range conf.GC.Policy {
+				b, err := units.RAMInBytes(p.KeepStorage)
+				if err != nil {
+					return nil, err
+				}
+				if b == 0 {
+					b = defaultKeepStorage
+				}
+				gcPolicy[i], err = toBuildkitPruneInfo(types.BuildCachePruneOptions{
+					All:         p.All,
+					KeepStorage: b,
+					Filters:     p.Filter,
+				})
+				if err != nil {
+					return nil, err
+				}
+			}
+		} else {
+			gcPolicy = mobyworker.DefaultGCPolicy(root, defaultKeepStorage)
```

Comment:
Why not initializing `gcPolicy` to this directly instead of having an else statement? then if the user wants to set it the `make` at 194 can override it